node/dataSources/catalog.ts

```diff
@@ -23,6 +24,10 @@ const memoryCache = new LRUCache<string, any>({max: 2000})
 
 metrics.trackCache('catalog', memoryCache)
 
+const inflightKey = ({baseURL, url, params, headers}: RequestConfig) => {
+  return baseURL! + url! + stringify(params, {arrayFormat: 'repeat', addQueryPrefix: true}) + `&segmentToken=${headers['x-vtex-segment']}`
```

Comment:
Don't you need a `?` separator anywhere? Does the `url` always have some querystring or the stringify include that somehow?

I also have no idea what this `stringify` is doing =P

Reply:
This inflight key is just for caching purposes. It could be `banana` :p it just needs to contain all the juicy bits. FWIW, the `addQueryPrefix` part adds the `?` separator.